### PR TITLE
[REVIEW-READY] removes simulated turfs (and asteroid station)

### LINKED ISSUE #16061: ~~~ SEMI-FREEZE UNDERWAY; READ DESCRIPTION ~~~
In #16026 I'm about to begin my efforts of repathing every turf in the code. Because of the magnitude of this change, I'm effecting a semi freeze (with Cheri's permission) to save me from the hellhole of conflicts that would otherwise arise. 
- No new changes to existing maps (open prs are unaffected) will be accepted. 
  - ~~Entirely new maps will be accepted, but further changes beyond the initial pr will not.~~
- Changes to existing turfs may or may not be accepted, at my discretion. 
  - Any path changes to turfs (including additions) will not be accepted. 

Sorry if this inconveniences anyone, but there's just too much for me to change without effecting this.

### PR DESCRIPTION
**Pull request by duncathan**
Disclaimer: this PR removes asteroid station. Reasons include:
- Ass is unmaintained
- Ass is incompatible with TGM due to a byond bug
- Ass is incompatible with lavaland
- It would take another hour of work to update ass to work properly with this PR and frankly with the other reasons there already I seriously am not feeling like putting the effort in

If somebody wants to put in the work to fix Ass themselves, you're more than welcome to - ping me in #coderbus (I'll be there as Dunc|Mobile or Dunc2) and I can let you know what needs to be done.

---

oh god this is gonna be painful

ok so here's the plan:
- [x] kill the `turf/simulated` path
- [x] make space turfs have one static, immutable (effectively) gas mixture
- [x] use our newfound path freedom to move vars for closed turfs to `turf/wall` and vars for open turfs to `turf/open` or something
- [x] deal with the infinite merge conflicts this will give me
- [x] test the fuck out of it

Closes #16061

### CODE REVIEW

**Comment by ChangelingRain** [comment]
Amazing.

**Comment by GunHog** [comment]
What benefit does this provide?

**Comment by duncathan** [comment]
@GunHog
- kills snowflake handling of space turfs by LINDA
- minimizes turf vars that go unused by a particular type
  - for instance, `air` is a var only used by open turfs, but still held by all simulated turfs. this change (does not yet, but) will make `air` only held by all open turfs
- removes the redundancy of `turf/simulated` because there are only two other children of turf, both of which can handle themselves (as with my space changes already done)

**Comment by phil235** [comment]
How are we even supposed to review a PR with 100k line changes?

**Comment by optimumtact** [comment]
git remote add dunc https://github.com/duncathan/-tg-station
git fetch --all
git checkout master
git merge upstream/master
git push

git diff dunc/turf -- code/

you can also git diff dunc/turf -- code/game/turfs/

it would help if his codebase was up to date

**Comment by ChangelingRain** [comment]
[It's not actually 100k, it's like 50k and a big chunk is maps.](https://github.com/tgstation/-tg-station/pull/16026/files?w=0)

**Comment by duncathan** [comment]
I fucked something up; box has about 60000 whitespace changes. But yea the vast majority of the pr is maps, because just about every map instance has to be changed.

**Comment by GunHog** [comment]
>How are we even supposed to review a PR with 100k line changes?

It is hilarious that you are the one to say this!

**Comment by lzimann** [comment]
i think someone is going insane, good luck

**Comment by Miauw62** [comment]
absolute madman

**Comment by kyrahabattoir** [comment]
So non open turfs will only track thermal energy going through them right?

**Comment by duncathan** [comment]
Well technically it's temperature they track, but thermal energy is easily calculated by multiplying temperature and heat capacity.

**Comment by KazeEspada** [comment]
Suggested repaths:
Plating path really means: You can put a tile on it. Not sure what to call it though. turf/open/base?
Asteroid path really means: You can hit it with a shovel to get sand. Should probably be called digable. turf/open/base/digable?

**Comment by KorPhaeron** [comment]
@duncathan 

Can you please repath lava to something other than plating while you are doing this? That was a silly idea on my part

**Comment by Bawhoppen** [comment]
how much longer on this shit I want to change maps

**Comment by duncathan** [comment]
I hope to be finished by the end of next week

**Comment by Bawhoppen** [comment]
@dunc when I click the files changed it literally lags, you maniac

**Comment by duncathan** [comment]
it's only gonna get worse
not too much worse, but still pretty bad

**Comment by PKPenguin321** [comment]
400k line changes
jesus dicks

**Comment by duncathan** [comment]
this is why we have mapping freezes, kids

**Comment by duncathan** [comment]
it compiles

**Comment by lzimann** [comment]
conflict fixing time!

**Comment by ChangelingRain** [comment]
oh, but they are fixed,

**Comment by Bawhoppen** [comment]
still says theres conflicts

**Comment by duncathan** [comment]
This is ready to review. Only concerns I have with it right now are the increases in roundstart setup time for the Lighting SS and the Air SS to about 70-90 seconds each. Not sure what the best way to deal with that is, so I'm open to ideas.

**Comment by duncathan** [comment]
@tgstation/commit-access 
Ready for review. Good luck with that :B

**Comment by optimumtact** [comment]
merge conflicts

**Comment by KorPhaeron** [comment]
+254,887 −263,059 

Let us just merge it instead, the players can review it for us

**Comment by Bawhoppen** [comment]
that 90 second wait is probably not good though kor

**Comment by duncathan** [comment]
yeah right now roundstart setup takes about 3 and a half minutes which is less than ideal.
again, if anyone's got some great ideas for that, I'm all ears.

**Comment by optimumtact** [comment]
did you make sure starlight isn't proccing on the newly non simulated simulated turfs?

**Comment by phil235** [comment]
merge conflicts

**Comment by KorPhaeron** [comment]
Should we stop merging things for a couple days to get this in first?

**Comment by lzimann** [comment]
FEATURE F R E E Z E V2

**Comment by optimumtact** [comment]
Yes feature freeze for this seems fair

**Comment by KorPhaeron** [comment]
@duncathan 

Maybe (if you know off the top of your head) a list of the files with the important changes that need to be reviewed instead of poking for them among 265 would make it easier/faster for maintainers

**Comment by optimumtact** [comment]
Lets review this by merging it and smoothing out any issues that arise.

**Comment by duncathan** [comment]
- `air.dm`
- `code/game/turfs`
- `code/modules/atmospherics/environmental`
- `code/modules/atmospherics/gasmixtures`
- `lighting_system.dm`
- `mine_turfs.dm`

@tgstation/commit-access files and folders that are actually in need of review. everything else is just find-replace sort of thing.

**Comment by duncathan** [comment]
I'm not gonna fix the conflicts on this until someone's willing to merge, by the way

**Comment by phil235** [comment]
@duncathan In code/module/mining/mine_turfs.dm you removed the name of /turf/closed/mineral/iron and all other deposits, except gibtonite. So now all these deposits are just called "rock". Is that intended?

(compare https://github.com/duncathan/-tg-station/blob/turfs/code/modules/mining/mine_turfs.dm#L105 and https://github.com/tgstation/-tg-station/blob/0288f7c592cb2784c45ce7be61ab31c6e65c2090/code/modules/mining/mine_turfs.dm#L148)

**Comment by KorPhaeron** [comment]
@duncathan 

Maybe we should organize a time where you fix the conflicts and I testmerge it to see how it handles on server.

@phil235 

That's fine. The most recent master has the names removed as well.

**Comment by phil235** [comment]
Why the difference between https://github.com/duncathan/-tg-station/blob/turfs/code/modules/lighting/lighting_system.dm#L316 and https://github.com/tgstation/-tg-station/blob/master/code/modules/lighting/lighting_system.dm ?

**Comment by duncathan** [comment]
The return value is never used, so i made it not return anything. `update_starlight()` already checks for config.starlight. That was done in an attempt to speed up roundstart, and it did shave off a couple seconds. It'll slow roundstart down if there's no starlight, but I'd rather prioritize the config that we use.

File: code/controllers/subsystem/air.dm [hunk 1]
```diff
@@ -122,12 +122,12 @@ var/datum/subsystem/air/SSair
 
 
 /datum/subsystem/air/proc/process_high_pressure_delta()
-	for(var/turf/T in high_pressure_delta)
+	for(var/O in high_pressure_delta)
+		var/turf/open/T = O
```
Comment (phil235) [inline code/controllers/subsystem/air.dm]: here you're assuming that all elements in the high_pressure_delta list is a /turf/open, but that list is defined above as var/list/turf/high_pressure_delta, not var/list/turf/open/high_pressure_delta. Which is right?

Comment (duncathan) [inline code/controllers/subsystem/air.dm]: turf/open. The definition is a leftover from me removing turf/simulated. Will fix.

**Comment by Cheridan** [comment]
H-HERE I GO.

**Comment by duncathan** [comment]
i'm free.

we're all free.

**Comment by GunHog** [comment]
You are not free until you fix all the bugs and exploits our wonderful QA team (the players) find and report!

**Comment by Ultimate-Chimera** [comment]
This is broken. ;-;

**Comment by optimumtact** [comment]
Please report bugs via the issue tracker

**Comment by Ultimate-Chimera** [comment]
Actually, this works fine with 510. I was using 509.

**Comment by optimumtact** [comment]
@Ultimate-Chimera are you running windows and linux, and if linux, did you have bygex installed?

**Comment by Ultimate-Chimera** [comment]
@optimumtact  Windows.

**Comment by Incoming5643** [comment]
Literally every round is ending in a hyper pressurized Armageddon, can we revert this already?

**Comment by optimumtact** [comment]
no

**Comment by Kevak** [comment]
I desire pictures of the mapchange.

**Comment by optimumtact** [comment]
This didn't have any visible changes to maps, it just rewrote underlying atmos turf technology

**Comment by KorPhaeron** [comment]
it ruined the game don't port it

**Comment by SamHPurp** [comment]
God help me, I'm considering porting this to Para.

**Comment by duncathan** [comment]
@SamHPurp find me on #coderbus (Dunc or Dunc|Mobile) if you need help, which I'm sure you will. This task is a monster

**Comment by SamHPurp** [comment]
@duncathan I'm keen to get Lavaland ported, and this is the major barrier in us doing so.

**Comment by KorPhaeron** [comment]
@SamHPurp 

You'd probably be better off just rebasing and then readding your features than trying to catch up on years of turf/atmos/lighting changes for porting large projects

**Comment by KorPhaeron** [comment]
This PR was pretty broken and required a ton of follow up

**Comment by SamHPurp** [comment]
@KorPhaeron confidence inspiring

**Comment by KorPhaeron** [comment]
Not trying to stop you since getting lavaland on every server would be a point of pride for me, just trying to set realistic expectations. 2 years of refactors to the basic building blocks of the game ported piece by piece sounds like a fucking nightmare

**Comment by SamHPurp** [comment]
@KorPhaeron Aye. Sounds painful and tedious. Not impossible, I just need to define a healthy workflow of Paracode, the PRs and current TG code I think.....

**Comment by optimumtact** [comment]
You can do it, if you believe in yourself

**Comment by LemonInTheDark** [comment]
he couldn't do it

**Comment by duncathan** [comment]
can't blame him tbh

**Comment by Kiyahitayika** [comment]
> God help me, I'm considering porting this to Para.

God help me, I'm considering porting this to Scorpio, a Paradise downstream server.

**Comment by Kiyahitayika** [comment]
You don't need a healthy workflow, just an unhealthy habit of using coding as stress-relief ;--;

**Pull request merged**